src/commands.ts

```diff
@@ -21,59 +21,71 @@ export function registerCommands() {
         const options = <QuickPickOptions> {
             placeHolder: "Select Action"
         };
-    
+
         window.showQuickPick(items, options).then(selection => {
             if (typeof selection === "undefined") {
                 return;
             }
-            
+
             if (selection.label === "File Access: Make Read Only") {
                 updateFileAccess(FileAccess.ReadOnly);
             } else {
                 updateFileAccess(FileAccess.Writeable);
             }
         });
     }
-    
+
     function indicatorAction() {
         const action: string = workspace.getConfiguration("fileAccess").get("indicatorAction");
         switch (action) {
             case "toggle":
-                if (Operations.isReadOnly(window.activeTextEditor.document)) {
-                    updateFileAccess(FileAccess.Writeable);
-                } else {
-                    updateFileAccess(FileAccess.ReadOnly);
-                }
+                toggleFileAccess();
                 break;
-    
+
             case "choose":
                 changeFileAccess();
                 break;
-    
+
             default:
                 console.log(`Received bad action '${action}'`);
                 window.showErrorMessage(`No indicator action '${action}' is available`);
         }
     }
-    
+
+    function toggleFileAccess() {
+        if (!window.activeTextEditor) {
+            window.showInformationMessage("Open a file first to update it attributes");
+            return;
+        }
```

Comment:
Reuse the two validations from `updateFileAccess` method in `operations.ts`. But instead of copying here the other one, just create a new function in `operations.ts` and exports. 

Something like: 

```ts
    export function isValidDocument(textEditor: TextEditor | undefined): boolean {
        if (!textEditor) {
           window.showInformationMessage("Open a file first to update it attributes");
           return resolve (false);
       }
            
       if (textEditor.document.uri.scheme === "untitled") {
           window.showInformationMessage("Save the file first to update it attributes");
           return resolve (false);
       }
    }
```